contracts/FlowStakingCollection.cdc

```diff
@@ -0,0 +1,844 @@
+                    self.depositTokens(from: <-tokens)
+                } else {
+                    let staker = self.tokenHolder!.borrow()!.borrowStaker()
+                    staker.withdrawUnstakedTokens(amount: amount)
+                }
+            }
+        }
+
+        /// Function to withdraw rewarded tokens for an existing Stake or Delegation record in the StakingCollection
+        pub fun withdrawRewardedTokens(nodeID: String, delegatorID: UInt32?, amount: UFix64) {
+            pre {
+                self.doesStakeExist(nodeID: nodeID, delegatorID: delegatorID): "Specified stake does not exist"
+            }
+
+            if let _delegatorID = delegatorID {
+                if let delegator = self.borrowDelegator(nodeID, _delegatorID) {
+                    // We update the unlocked tokens used field before withdrawing because 
+                    // rewards are newly minted and not immediately reflected in the tokens used fields
+                    self.unlockedTokensUsed = self.unlockedTokensUsed + amount
+
+                    let tokens <- delegator.withdrawRewardedTokens(amount: amount)
+
+                    self.depositTokens(from: <-tokens)
+                } else {
+                    let delegator = self.tokenHolder!.borrow()!.borrowDelegator()
+                    
+                    delegator.withdrawRewardedTokens(amount: amount)
+
+                    // move the unlocked rewards from the locked account to the unlocked account
+                    let unlockedRewards <- self.tokenHolder!.borrow()!.withdraw(amount: amount)
+                    self.unlockedVault.borrow()!.deposit(from: <-unlockedRewards)
+                }
+            } else {
+                if let node = self.borrowNode(nodeID) {
+                    self.unlockedTokensUsed = self.unlockedTokensUsed + amount
+
+                    let tokens <- node.withdrawRewardedTokens(amount: amount)
+
+                    self.depositTokens(from: <-tokens)
+                } else {
+                    let staker = self.tokenHolder!.borrow()!.borrowStaker()
+                    
+                    staker.withdrawRewardedTokens(amount: amount)
+
+                    // move the unlocked rewards from the locked account to the unlocked account
+                    let unlockedRewards <- self.tokenHolder!.borrow()!.withdraw(amount: amount)
+                    self.unlockedVault.borrow()!.deposit(from: <-unlockedRewards)
+                }
+            }
+        }
+
+        // Closers
+
+        /// Closes an existing stake or delegation, moving all withdrawable tokens back to the users account and removing the stake
+        /// or delegator object from the StakingCollection.
+        pub fun closeStake(nodeID: String, delegatorID: UInt32?) {
+            pre {
+                self.doesStakeExist(nodeID: nodeID, delegatorID: delegatorID): "Specified stake does not exist"
+            }
+
+            if let _delegatorID = delegatorID {
```

Comment:
Maybe just shadow the optional variable with the non-optional one, i.e. name it the same:

```suggestion
            if let delegatorID = delegatorID {
```